database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala

```diff
@@ -81,6 +81,16 @@ trait WorkflowMetadataSummaryEntryComponent {
       } yield summary.workflowExecutionUuid).take(batchSize)
     })
 
+  val countRootWorkflowIdsByArchiveStatusAndEndedOnOrBeforeThresholdTimestamp = Compiled(
+    (metadataArchiveStatus: Rep[Option[String]], workflowEndTimestampThreshold: Rep[Timestamp]) => {
+      (for {
+        summary <- workflowMetadataSummaryEntries
+        if summary.rootWorkflowExecutionUuid.isEmpty && summary.parentWorkflowExecutionUuid.isEmpty // is root workflow entry
+        if summary.metadataArchiveStatus === metadataArchiveStatus
+        if summary.endTimestamp <= workflowEndTimestampThreshold
+      } yield summary.workflowExecutionUuid).length
+    })
```

Comment:
There is already an index on metadata archive status but currently no index on end timestamp. So it seems that if there are a large number of Archived workflows this query could do a lot of scanning, not unlike what we saw with the backed up summarization queue.

Reply:
...although I'm not sure how much an end timestamp index would help with `<=` 🙁 . Maybe a config switch to turn off calculating this metric if we know things are backed up (like yesterday)?

Reply:
I think it's fine that the metric is potentially slow, my request would be that it should not happen in serial with the actual processing of metadata deletion.

I would say:

* If the number of metadata entries returned by the deletion processing is less than the maximum batch size, we know exactly how many there are needing deletion so we don't need to re-run this.
* If the number of metadata entries returned by the deletion processing query is equal to the maximum batch size, we should activate a parallel action to count how many there are, and continue polling at some interval until the backlog falls back to zero.